rest_framework_json_api/filters/filter.py

```diff
@@ -0,0 +1,135 @@
+            """
+            complain that they need django-filter
+            TODO: should this be a warning or an exception?
+            """
+            warnings.warn("must install django-filter package to use JSONAPIDjangoFilter")
+
+        def filter_queryset(self, request, queryset, view):
+            """
+            do nothing
+            """
+            return queryset
+
+else:
+    class JSONAPIDjangoFilter(DjangoFilterBackend):
+        """
+        A Django-style ORM filter implementation, using `django-filter`.
+
+        This is not part of the jsonapi standard per-se, other than the requirement
+        to use the `filter` keyword: This is an optional implementation of style of
+        filtering in which each filter is an ORM expression as implemented by
+        DjangoFilterBackend and seems to be in alignment with an interpretation of
+        http://jsonapi.org/recommendations/#filtering, including relationship
+        chaining. It also returns a 400 error for invalid filters.
+
+        Filters can be:
+        - A resource field equality test:
+            `?filter[qty]=123`
+        - Apply other [field lookup](https://docs.djangoproject.com/en/stable/ref/models/querysets/#field-lookups)  # noqa: E501
+          operators:
+            `?filter[name.icontains]=bar` or `?filter[name.isnull]=true...`
+        - Membership in a list of values:
+            `?filter[name.in]=abc,123,zzz (name in ['abc','123','zzz'])`
+        - Filters can be combined for intersection (AND):
+            `?filter[qty]=123&filter[name.in]=abc,123,zzz&filter[...]`
+        - A related resource path can be used:
+        `?filter[inventory.item.partNum]=123456 (where `inventory.item` is the relationship path)`
+
+        If you are also using rest_framework.filters.SearchFilter you'll want to customize
+        the name of the query parameter for searching to make sure it doesn't conflict
+        with a field name defined in the filterset.
+        The recommended value is: `search_param="filter[search]"` but just make sure it's
+        `filter[<something>]` to comply with the jsonapi spec requirement to use the filter
+        keyword. The default is "search" unless overriden but it's used here just to make sure
+        we don't complain about it being an invalid filter.
+        """
+        # TODO: find a better way to deal with search_param.
+        search_param = api_settings.SEARCH_PARAM
+
+        # Make this regex check for 'filter' as well as 'filter[...]'
+        # Leave other incorrect usages of 'filter' to JSONAPIQueryValidationFilter.
+        # See http://jsonapi.org/format/#document-member-names for allowed characters
+        # and http://jsonapi.org/format/#document-member-names-reserved-characters for reserved
+        # characters (for use in paths, lists or as delimiters).
+        # regex `\w` matches [a-zA-Z0-9_].
+        # TODO: U+0080 and above allowed but not recommended. Leave them out for now. Fix later?
```

Comment:
I think this is fine for now to be more restrictive. Can always be relaxed at a later point.